libyul/Object.h

```diff
@@ -88,9 +88,10 @@ struct ObjectDebugData
 /**
  * Yul code and data object container.
  */
-struct Object: public ObjectNode
+class Object: public ObjectNode
 {
 public:
+	explicit Object(Dialect const&) {}
```

Comment:
Why isn't the dialect being stored?

Reply:
I got a bit confused by it too, but I think it's because it is actually assigned and worked with in # 15347...

Reply:
Probably, but I don't see why we can't have the complete change here. Making stuff use the dialect from `Object` does not seem like it should depend on the changes we're making to Yul AST in #15347. I'm usually for splitting things more, but IMO this one is missing too much context.

Reply:
Because the full change means that AsmPrinter and AsmJsonConverter also get the dialect and do something with it. Right now they don't do anything with it, so it's a matter of cutoff IMO. But sure, I can add the respective constructor arguments there and the rest comes in the follow up PR. Or we just scrap this one.